integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonAnalysisRules.scala

```diff
@@ -143,52 +168,63 @@ case class CarbonIUDAnalysisRule(sparkSession: SparkSession) extends Rule[Logica
       selectPlan
     }
     val finalPlan = if (filter.length > 0) {
-      val alias = table.alias.getOrElse("")
       var transformed: Boolean = false
       // Create a dummy projection to include filter conditions
       var newPlan: LogicalPlan = null
       if (table.tableIdentifier.database.isDefined) {
         newPlan = parser.parsePlan("select * from  " +
-           table.tableIdentifier.database.getOrElse("") + "." +
-           table.tableIdentifier.table + " " + alias + " " + filter)
+                                   table.tableIdentifier.database.getOrElse("") + "." +
+                                   table.tableIdentifier.table + " " + alias.getOrElse("") + " " +
+                                   filter)
       }
       else {
         newPlan = parser.parsePlan("select * from  " +
-           table.tableIdentifier.table + " " + alias + " " + filter)
+                                   table.tableIdentifier.table + " " + alias.getOrElse("") + " " +
+                                   filter)
       }
       newPlan transform {
-        case UnresolvedRelation(t, Some(a))
-          if !transformed && t == table.tableIdentifier && a == alias =>
+        case CarbonUnresolvedRelation(t)
+          if !transformed && t == table.tableIdentifier =>
           transformed = true
-          // Add the filter condition of update statement  on destination table
-          SubqueryAlias(alias, updatedSelectPlan, Option(table.tableIdentifier))
+
+          val subqueryAlias = CarbonClassReflectionUtils
+            .getSubqueryAlias(sparkSession, alias, updatedSelectPlan, Some(table.tableIdentifier))
+          subqueryAlias
```

Comment:
Don't assign to the variable, just return it.  And format the code as below.
```
CarbonClassReflectionUtils.getSubqueryAlias(
              sparkSession, 
              alias, 
              updatedSelectPlan, 
              Some(table.tableIdentifier))
```

Reply:
Done